### ISSUE TITLE
Regression: Mobile Suit Gundam Unicorn RE:0096 not parsing again

### ISSUE THREAD

**Issue by Thraxis** [pymedusa/SickRage#1027]
### Before submitting your issue:

Enable debug logging in SickRage settings, reproduce the error (be sure to disable after the bug is fixed)

```
Medusa Info:	Branch: develop
Commit: 81dd9b4f83003a4309aeb8210a9191cfdaf1756a
Version: v0.1.3
 Python Version:	2.7.10 (default, Jun 26 2015, 16:00:40) [GCC 4.8.2]
 SSL Version:	OpenSSL 1.0.1o 12 Jun 2015
 OS:	Linux-4.1.18-unRAID-x86_64-AMD_C-60_APU_with_Radeon-tm-_HD_Graphics-with-slackware-14.1
 Locale:	en_US.UTF-8
 	 
 	 
 User:	nobody
 Program Folder:	/usr/local/Sickrage
 Config File:	/mnt/cache/appdata/sickrage/config.ini
 Database File:	/mnt/cache/appdata/sickrage/sickbeard.db
 Cache Folder:	/mnt/cache/appdata/sickrage/cache
 Log Folder:	/mnt/cache/appdata/sickrage/Logs
 Arguments:	['--daemon', '--port', '8082', '--datadir', '/mnt/cache/appdata/sickrage', '--pidfile', '/var/run/Sickrage/Sickrage.pid']
```

```
2016-09-04 12:39:28 DEBUG    SEARCHQUEUE-MANUAL-142601 :: [NyaaTorrents] :: [81dd9b4] Unable to match [HorribleSubs].Mobile.Suit.Gundam.Unicorn.RE.0096.-.21.[480p].mkv to a show in your database. Parser result: release_group: HorribleSubs, title: Mobile Suit Gundam Unicorn RE, episode: [], absolute_episode: [21, 96], screen_size: 480p, container: mkv, type: episode, quality: Unknown, season: None

```

**Comment by ratoaq2** [comment]
That happens if there's no exception list for that show:

The show name in your list should be something like:
`Mobile Suit Gundam UC RE:0096`

And you should have at least this exception:
`Mobile Suit Gundam Unicorn RE 0096`

I just tested that file in our test suite:
```
? "[GROUP].Mobile.Suit.Gundam.Unicorn.RE.0096.-.14.[720p].mkv"
: title: Mobile Suit Gundam Unicorn RE 0096
  absolute_episode: 14
  episode: 14
  screen_size: 720p
  release_group: GROUP
  container: mkv
  mimetype: video/x-matroska
  type: episode
```

Has any config changed for this show in your setup?

**Comment by Thraxis** [comment]
Only changes I have made have been updating to the current dev branch.
```
 Internal name cache for Mobile Suit Gundam UC RE:0096 set to: [ mobile suit gundam uc re0096, mobile suit gundam unicorn re0096, kidou senshi gundam unicorn, mobile suit gundam unicorn re 0096 ]
```

**Comment by ratoaq2** [comment]
That wasn't change since the unicorn was fixed more than a month ago.

**Comment by ratoaq2** [comment]
There could be only 1 thing:

Could you remove this line and try it again?

https://github.com/pymedusa/SickRage/blob/develop/sickbeard/name_parser/guessit_parser.py#L102
`@region.cache_on_arguments(expiration_time=EXPECTED_TITLES_EXPIRATION_TIME)`

**Comment by Thraxis** [comment]
Well this is a pain in the ass.

Not sure why this behavior happened but it caused cache contamination.

Mobile Suit Gundam Unicorn did not have an absolute episode number set.  I did a manual search and was getting the parse error I posted above.

I restarted and did a force update on the show, it updated and had the absolute episode number (episode 21) at that time and was then parsing correctly.  BUT the cache was contaminated because the episodes it parsed when the absolute number was missing had no episode listed in the cache.

I manually deleted all records in the cache tables that had '||' for the episodes field and now it's picking everything up fine with a manual search.

**Comment by ratoaq2** [comment]
You did reset the db cache, but did you change the code (removed that line)?  or those values were cached in db sometime before when we first had that issue?

**Comment by Thraxis** [comment]
I did not change the code.
The values were cached when I did a manual search (which is when I posted this issue as the name parse was wrong).

I am not sure why but it seems in the manual search with the wrong absolute number it parsed the name incorrectly, this resulted in there being cache contamination which meant when the absolute number was available and it was parsing correctly those items were skipped over as they were cached but did not have an episode number available.
